azurerm/internal/services/web/tests/resource_arm_app_service_hybrid_connection_test.go

```diff
@@ -0,0 +1,147 @@
+		CheckDestroy: testCheckAzureRMAppServiceHybridConnectionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMAppServiceHybridConnection_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAppServiceHybridConnectionExists(data.ResourceName),
+				),
+			},
+		},
+	})
+}
+
+func testCheckAzureRMAppServiceHybridConnectionDestroy(s *terraform.State) error {
+	client := acceptance.AzureProvider.Meta().(*clients.Client).Web.AppServicesClient
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_app_service_hybrid_connection" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resGroup := rs.Primary.Attributes["resource_group_name"]
+		namespaceName := rs.Primary.Attributes["namespace_name"]
+		relayName := rs.Primary.Attributes["relay_name"]
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+		resp, err := client.GetHybridConnection(ctx, resGroup, name, namespaceName, relayName)
+
+		if err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return nil
+			}
+
+			return err
+		}
+
+		return nil
+	}
+	return nil
+}
+
+func testCheckAzureRMAppServiceHybridConnectionExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %s", resourceName)
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resGroup := rs.Primary.Attributes["resource_group_name"]
+		namespaceName := rs.Primary.Attributes["namespace_name"]
+		relayName := rs.Primary.Attributes["relay_name"]
+
+		conn := acceptance.AzureProvider.Meta().(*clients.Client).Web.AppServicesClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+		resp, err := conn.GetHybridConnection(ctx, resGroup, name, namespaceName, relayName)
+		if err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: App Service Hybrid Connection %q (resource group: %q) does not exist", name, resGroup)
+			}
+
+			return fmt.Errorf("Bad: GetHybridConnection on appServicesClient: %+v", err)
+		}
+
+		return nil
+	}
+}
+
+func testAccAzureRMAppServiceHybridConnection_basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_app_service_plan" "test" {
+  name                = "acctestASP-%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+
+  sku {
+    tier = "Standard"
+    size = "S1"
+  }
+}
+
+resource "azurerm_app_service" "test" {
+  name                = "acctestAS-%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  app_service_plan_id = azurerm_app_service_plan.test.id
+}
+
+resource "azurerm_relay_namespace" "test" {
+  name                = "acctestrn-%d"
```

Comment:
Could we try and use capitals here (if the resource allows it) - this is to ensure the API handles casing correctly.
```suggestion
  name                = "acctest-RN-%d"
```